nameko_stripe/pytest.py

```diff
@@ -0,0 +1,70 @@
+import json
+import time
+
+import pytest
+import stripe
+
+from nameko_stripe import constants
+
+
+@pytest.fixture
+def stripe_config(web_config):
+    config = web_config.copy()
```

Comment:
Stripe webhook handlers are built on top of http handlers that's why besides secrets they also need web config. The dependency provider doesn't need web config, but it also doesn't need endpoint secret. Entrypoints on the other hand need endpoint secret and web setup. It seems better to provide all of them out of the box without asking users of the library to define all these keys on their own when testing their services.

Reply:
One of the nice things about fixtures is that we can leave them configurable and choose which ones to use in any case.

It's true that the dependency provider does not need the endpoint secret either but this seems different to coupling this fixture to the `web_config`, which has nothing to do with the config. I see these as two different things.

It seems natural to have `stripe_config` as a fixture that returns all the Stripe config. So that clients using it can choose what to get from it (maybe all of it). It does not seem a good approach to have a few different `stripe_config` fixtures returning different parts of the config. All the tests should pass with the config keys that are supported by the library and if any values need to be changed then other fixtures can override them.

A different thing seems the approach saying that every time we use `stripe_config` we should also execute and use `web_config`, which is not true for the dependency provider in this library and also not the case for how we use the library at the moment.

I do not mind having all the library configuration returned by this fixture but I'd like to have the freedom to choose when to execute and import the rest of the fixtures.

By having the fixtures configured like this, you have the freedom of choosing which one(s) you want to use|: just `stripe_config`, `config` with all the configuration, just `web_config`...

```python
# pytest.py

def stripe_config()
    return {
        "STRIPE": {
            "SECRET_KEY": "sk_test_EWX...FbN",
            "ENDPOINT_SECRET": "whsec_6Yi...Igu",
            "LOG_LEVEL": "debug",
        }
    }
``` 

```python
# conftest.py

def config(web_config, stripe_config)
    config = web_config.copy()
    config.update(stripe_config)
    return config
```